Sources/SourceControl/GitRepository.swift

```diff
@@ -600,6 +600,17 @@ private class GitFileSystemView: FileSystem {
         }
     }
     
+    func isExecutableFile(_ path: AbsolutePath) -> Bool {
+        do {
+            if let entry = try getEntry(path), entry.type == .executableBlob {
+                return true
+            }
+            return false
+        } catch {
+            return false
+        }
```

Comment:
the 3 preceding bits of logic all follow that same pattern - update them all to keep it consistent? (`isDirectory`, `isFile`, and `isSymlink`) - use the `try?` on all of them instead of the longer `do {} catch {}` block structure?